src/visitable.cpp

```diff
@@ -72,21 +72,30 @@ bool visitable<T>::has_item_with( const std::function<bool( const item & )> &fil
     } ) == VisitResponse::ABORT;
 }
 
+// Sums the two terms, being careful to not trigger any overflows.
+// Assumes that all terms are positive and base <= limit.
+static long accumulate_with_limit( long base, long addend, long limit ) {
```

Comment:
Did you mean I should just clamp to numeric_limits<int>::max()?  If so ok I've come around to agreeing with that.